srcalc/src/main/webapp/js/enterVariables.js

```diff
@@ -1,78 +1,72 @@
 /* JavaScript code just for the enterVariables page */
 
-$(document).ready(function(){
+// We're about to replace the procedureSelectGroup with a jQuery UI
+// dialog. Insert a hidden input and a textual display as the target
+// of the user selection from the dialog.
+// Get the procedure hidden input, if it exists.
+function onSelectProcedure(cptCode, displayString) {
+	// Get the selected button's value and change the display string
+	var hiddenInput = $('.procedureHiddenInput');
+    hiddenInput.val(cptCode);
+    var userDisplay = $('.procedureDisplay');
+    userDisplay.html(displayString);
+    $(".procedureSelectGroup").dialog("close");
+    // Return false so there is no redirect after clicking
+    return false;
+}
+
+/**
+ * Moves the procedureSelectGroup table into a jQuery UI dialog.
+ */
+function initProcedureSelect() {
+    // Note that this code assumes there is only one procedureSelectGroup
+    // on the page.
+
+    // Lookup the procedure variable name and the initially-selected procedure.
+    var hiddenInput = $('.procedureHiddenInput');
+    var varName = hiddenInput.attr('name');
+    var selectedProcedure = hiddenInput.val();
+    var procedureSelectDialog = $(".procedureSelectGroup").dialog({
+        autoOpen: false,
+        width: 700,   // body with is 768px
+        modal: true
+    });
+    
 	// Set up the properties for the procedures DataTable
 	var proceduresTable = $("#procedureTable").dataTable({
 		"data": procedureArray,
 		"retrieve": true,
 		"deferRender": true,
-		"columnDefs":[
-		//Make the radio button column smaller so that IE
-		// will adjust column widths properly.
-		{
-			"width": "10%",
-			"targets": [0]
-		},
-		// Make the select column, description,  and RVU unsearchable
-		{
-			"bSearchable": false, 
-			"aTargets": [0,2,3]
-		}, {
-			// Make the select button unsortable
-			"bSortable": false,
-			"aTargets": [0]
-		}]
+		columns: [
+		          { data: 'cptCode' },
+		          { data: 'longDescription', searchable: false },
+		          { data: 'rvu', searchable: false },
+		          {
+		              data: 'cptCode',
+		              render: function (data, type, row) {
+		                  return '<a href="#" class="btn-link" name="' + varName +
+                              '" value="' + row.cptCode +
+                              '" data-display-string="' + row.displayString + '" ' +
+                              'onclick="onSelectProcedure(\''+ row.cptCode +'\',\'' + row.displayString +'\')">Select</a>';
```

Comment:
Using "onclick" is not recommended these days. It would be better to use jQuery's on() as so:

    $('a.selectProcedure').on('click', function() {
        var elem = $(this);
        // Store the CPT code and display string as "data" elements to
        // allow one callback to handle all selects.
        selectProcedure(elem.data('cpt-code'), elem.data('display-string');
    });